src/recap.js

```diff
@@ -122,65 +131,115 @@ function Recap() {
+        })
+        .then(data => fetch(`${SERVER_ROOT}recap/`, {
+          method: 'POST',
+          body: data,
+          headers: { 'Authorization': `Token ${N87GC2}`}
+        }))
+        .then(res => res.json())
+        .then( result => {
+          console.info(`RECAP: Successfully uploaded PDF: 'Success' ` +
+            `with processing queue id of ${result.id}`);
+          cb(result || null);
+          destroyTabStorage(cb.tab.id);
+        })
+        .catch(error => console.log(`RECAP: Error uploading PDF: ${error}`));
+    },
 
-        if (attachment_number && attachment_number !== '0'){
-          formData.append('attachment_number', attachment_number);
-        }
-        const ab = new Uint8Array(bytes[nonce])
-        const blob = new Blob([ab])
-        console.log("RECAP: Retrieved blob from storage", blob)
+    // Upload a zip file to the RECAP server, calling the cb with ok flag
+    uploadZipFile: (
+      pacer_court,
+      pacer_case_id,
+      cb
+    ) => {
+      console.info([
+        `RECAP: Attempting Zip upload to RECAP Archive with details:`,
+        `pacer_court: ${pacer_court}`,
+        `pacer_case_id: ${pacer_case_id}`,
+      ].join(' '));
 
-        formData.append('filepath_local', blob);
-        formData.append('upload_type', UPLOAD_TYPES['PDF']);
-        formData.append('debug', DEBUG);
+      // extract the tabId from the enhanced callback
+      // wait for chrome.storage.local to load the tabStorage
+      getItemsFromStorage(cb.tab.id)
+        .then(tabStorage => {
+          // create the formData
+          const ab = tabStorage['zip_blob'];
+          const docId = tabStorage['docId'];
+          const blob = blobFromArrayBuffer(ab);
 
-        $.ajax({
-          url: `${SERVER_ROOT}recap/`,
+          const formData = new FormData();
+          formData.append('court', PACER.convertToCourtListenerCourt(pacer_court));
+          pacer_case_id && formData.append('pacer_case_id', pacer_case_id);
+          formData.append('pacer_doc_id', docId);
+          formData.append('upload_type', UPLOAD_TYPES['ZIP']);
+          formData.append('debug', DEBUG);
+          formData.append('filepath_local', blob);
+          return formData;
+        })
+        .then(data => fetch(`${SERVER_ROOT}recap/`, {
           method: 'POST',
-          processData: false,
-          contentType: false,
-          data: formData,
-          success: function(data, textStatus, xhr){
-            console.info(`RECAP: Successfully uploaded PDF: '${textStatus}' ` +
-                        `with processing queue id of ${data['id']}`);
-            cb(data || null);
-          },
-          error: function(xhr, textStatus, errorThrown){
-            console.error(`RECAP: Ajax error uploading PDF. Status: ${textStatus}.` +
-                          `Error: ${errorThrown}`);
-          }
-        });
-
-      });
+          body: data,
+          headers: {'Authorization': `Token ${N87GC2}`}
```

Comment:
No. The browserSpecificFetch issue is because of a particular security restriction imposed by Firefox which prevents content scripts from calling fetch on a path ('/api/stuff') which we need to do to avoid hitting the "You are coming to Pacer from outside ... we will gouge you for monies". 

The recap functions are background functions that use the absolute recap URLs, so we can use the normal fetch mechanism.